src/modules/edc-demo/components/transfer-history/transfer-history-viewer.component.html

```diff
@@ -1,66 +1,89 @@
 <div id="wrapper">
-  <div *ngIf="transferProcesses$ | async as transferProcesses">
-    <button mat-raised-button color="primary" (click)="loadTransferProcesses()">
-      <mat-icon>cached</mat-icon>
-      Refresh
-    </button>
-    <mat-paginator
-      pageSize="transferProcesses.length"
-      hidePageSize="true"
-      [length]="transferProcesses.length"></mat-paginator>
+  <button mat-raised-button color="primary" (click)="loadTransferProcesses()">
+    <mat-icon>cached</mat-icon>
+    Refresh
+  </button>
+  <mat-paginator
+    *ngIf="transferProcessesList.isReady"
+    pageSize="transferProcessesList.data.transferProcesses.length"
+    hidePageSize="true"
+    [length]="
+      transferProcessesList.data.transferProcesses.length
+    "></mat-paginator>
+  <div class="flex flex-row justify-center">
+    <edc-demo-loading-state
+      *ngIf="transferProcessesList.isLoading"></edc-demo-loading-state>
+    <edc-demo-error-state
+      *ngIf="transferProcessesList.isError"
+      [data]="
+        transferProcessesList.error.failureMessage
+      "></edc-demo-error-state>
+    <edc-demo-empty-state
+      *ngIf="
+        transferProcessesList.isReady &&
+        transferProcessesList.data.transferProcesses.length === 0
```

Comment:
check which cardinalities are checked (in this entire page), fix number comparisons